src/plone.server/plone/server/traversal.py

```diff
@@ -69,7 +130,16 @@ def __init__(self, request, resource, view, rendered):
         self.rendered = rendered
 
     async def handler(self, request):
-        view_result = await self.view()
+        if request.method in WRITING_VERBS:
+            txn = request.conn.transaction_manager.begin(request)
+            async with locked(self.resource):
+                view_result = await self.view()
+            if txn is None:
+                await sync(request)(txn.commit)
+            else:
+                await sync(request)(txn.abort)
+        else:
+            view_result = await self.view()
```

Comment:
Because you can't pass the request to the transaction manager without storing it on the object with a with, no ?